tests/Helpers/LogAnalyticsHelper.cs

```diff
@@ -0,0 +1,206 @@
+                    Delay = TimeSpan.FromSeconds(2),
+                    MaxDelay = TimeSpan.FromSeconds(10)
+                }
+            };
+
+            _logsIngestionClient = new LogsIngestionClient(endpoint, _credential, options);
+            return _logsIngestionClient;
+        }
+        catch (Exception ex)
+        {
+            throw new InvalidOperationException($"Failed to create LogsIngestionClient: {ex.Message}", ex);
+        }
+        finally
+        {
+            _clientInitLock.Release();
+        }
+    }
+
+    /// <summary>
+    /// Creates and sends a log with the specified level and message.
+    /// </summary>
+    private async Task<HttpStatusCode> CreateAndSendLogAsync(
+        string level,
+        string message,
+        CancellationToken cancellationToken = default)
+    {
+        var workspaceId = await GetWorkspaceIdAsync().ConfigureAwait(false);
+
+        var log = new LogRecord
+        {
+            TimeGenerated = DateTimeOffset.UtcNow,
+            Level = level,
+            Message = message,
+            Application = "MonitorCommandTests"
+        };
+
+        return await SendLogAsync(workspaceId, [log], cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <summary>
+    /// Sends an information level log message.
+    /// </summary>
+    public Task<HttpStatusCode> SendInfoLogAsync(CancellationToken cancellationToken = default)
+    {
+        return CreateAndSendLogAsync(
+            "Information",
+            $"Test info message: {DateTimeOffset.UtcNow:O}",
+            cancellationToken);
+    }
+
+    /// <summary>
+    /// Sends both information and error test logs.
+    /// </summary>
+    public async Task<(HttpStatusCode infoStatus, HttpStatusCode errorStatus)> SendTestLogsAsync(
+        string testId,
+        CancellationToken cancellationToken = default)
+    {
+        var infoStatus = await CreateAndSendLogAsync(
+            "Information",
+            $"Test info message: {testId}",
+            cancellationToken).ConfigureAwait(false);
+
+        var errorStatus = await CreateAndSendLogAsync(
+            "Error",
+            $"Test error message {Guid.NewGuid()}",
+            cancellationToken).ConfigureAwait(false);
+
+        return (infoStatus, errorStatus);
+    }
+
+    /// <summary>
+    /// Sends log records to Azure Monitor using the Logs Ingestion API.
+    /// </summary>
+    private async Task<HttpStatusCode> SendLogAsync(
+        string customerId,
+        LogRecord[] logs,
+        CancellationToken cancellationToken = default)
+    {
+        var client = await GetLogsIngestionClientAsync(customerId).ConfigureAwait(false);
+
+        try
+        {
+            using var content = RequestContent.Create(logs);
+            _logger.LogInformation("Sending {Count} logs to workspace {WorkspaceId}", logs.Length, customerId);
+
+            cancellationToken.ThrowIfCancellationRequested();
+            var response = await client.UploadAsync(
```

Comment:
Will this return a response even if the service returned a non-2xx status or does it throw an exception here?

Reply:
Is there a question behind the question?  I'm not sure what you are getting at.

Reply:
In the `catch` block, we always return `HttpStatusCode.InternalServerError` for any exception. So, I wasn't sure what the behavior of `client.UploadAsync()` call would be if the service return other error codes like 401 or 403. Would that return a response or would that throw an exception? If it returns the response, then this looks good. If it throws an exception, we will be mapping all service errors to `InternalServerError`.

Reply:
Given this is a just a test helper I don't think it matters. Do you?

Reply:
For testing purposes, it shouldn't matter if we don't need to assert the status codes. Resolving the comment.